config/routes.js

```diff
@@ -1133,6 +1153,18 @@ module.exports = [
       },
     },
   },
+  {
+    method: 'DELETE',
+    path: '/api/v3/totp/device/{id}',
+    handler: TOTPController.destroyDevice,
```

Comment:
Shouldn't there be a check on the TOTP token (`AuthPolicy.isTOTPEnabledAndValid`) to remove a trusted device? We're asking for it to add a trusted device or to disable 2FA but not here.

Reply:
In my opinion, no. I looked online for some sort of guidance but I couldn't find anything definitive in terms of "set up 2FA like this" — however while I was documenting I did notice the same issue you're raising and I did think about the impact:

It's clear that creating the saved device is an important security event requiring TOTP. The holder of the second factor must opt out of requiring it for future operations.

But after the device is "untrusted," it means on that device you are once again required to submit TOTPs with every sensitive request. Removing a saved device is actually tightening security since the second factor must once again be at hand for any operation.

The saved devices expire automatically after 30 days, so it has a lot of parallels with a normal user session: we need a password to create the session, but ending the session can be done by anyone who clicks the logout button, no password needed.